modules/targetAcquisition/targetAcquisition.py

```diff
@@ -75,17 +89,19 @@ def get_coordinates(self, newFrame=np.zeros((416, 416, 3))):
         # If new frame has been specified (is non-empty, set the current frame to the given frame)
         if np.count_nonzero(newFrame) != 0:
```

Comment:
Don't know if this is the most effective implementation of this, now that I'm looking at it

Reply:
Ineffective in what sense?

Reply:
Everytime this function runs, this piece of code will go through the entire 416x416x3 or whatever image to count the number of zeroes in order to determine if the current frame is valid

Might be a tiny difference in performance but nonetheless this isn't an ideal way to achieve what this line of code is trying to do

Reply:
Performance wise its not an issue because numpy uses some voodoo parallelism under the hood, I suspect this function does boolean masking, but regardless when you use a built in numpy function to inspect an np array it typically doesn't iterate so performance is a non issue